File: nacos/3.2.3/analyzed/api/src/main/java/com/alibaba/nacos/api/naming/listener/Event.java
/*
 * Copyright 1999-2018 Alibaba Group Holding Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.alibaba.nacos.api.naming.listener;

/**
 * 命名服务变更事件标记接口。
 *
 * <p>具体事件类型如 {@code NamingEvent} 携带服务名与实例列表快照，由 {@link EventListener} 接收。</p>
 *
 * @author nkorange
 */
public interface Event {
    
}

File: harness/2.28.2/analyzed/core/event.go
// Copyright 2019 Drone IO, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package core

// 钩子事件类型常量，标识触发构建的 SCM 事件种类。
const (
	EventCron        = "cron"         // 定时任务触发
	EventCustom      = "custom"       // 自定义事件
	EventPush        = "push"         // 推送
	EventPullRequest = "pull_request" // Pull Request
	EventTag         = "tag"          // 标签
	EventPromote     = "promote"      // 晋升/晋级部署
	EventRollback    = "rollback"     // 回滚
)
